projects/ngx-charts-on-fhir/src/lib/fhir-data/fhir-data.service.ts

```diff
@@ -100,10 +112,10 @@ export class FhirDataService {
         "reference": `Patient/${this.client?.patient.id}`
       },
       "encounter": {
-        "reference": "Encounter/23249"
+        "reference": `Encounter/${this.client?.encounter.id}`
```

Comment:
Are you sure we will have an encounter in the context when launching a patient app? We should check if it's there first so we don't add a reference for `Encounter/undefined`

Reply:
Yes, We get encounter because we add encounter in environment.
![image](https://user-images.githubusercontent.com/117890382/217469483-38d1da63-edc9-47a3-b6e6-d1c78aab4a13.png)

Reply:
But that is a fake context that we use for testing. We can put whatever we want in there. The question is whether a real SMART launch for a patient app will include an encounter in the context. We don't need to answer that question here, but we do need to do some defensive coding to ensure that we are not saving FHIR resources that look like this:
```json
"encounter": {
  "reference": "Encounter/undefined"
}
```
Please test your code without an encounter in the context to make sure it still behaves sensibly.

Reply:
@semicolin, when I have tried to create a resource without an encounter then it gives 400, and in client, we get null as an encounter

Reply:
I remove the encounter from the resource and now it works without an encounter.